render_shapes crashes with AssertionError when all geometries are empty

# `render_shapes` crashes with `AssertionError` when all geometries are empty

**Environment**: `spatialdata-plot` `0.3.4.dev` (main, commit `5cfedc7`), Python 3.13

---

## Problem

When a shapes element contains only empty geometries (e.g., `Polygon()`), calling `render_shapes().show()` crashes with a bare `AssertionError`:

```
AssertionError: Cannot compute extent of an empty collection of geometries.
```

This occurs in `basic.py:1328` where `get_extent(sdata, coordinate_system=cs, ...)` is called. When all shapes are empty, `get_extent` raises an `AssertionError` rather than a clear `ValueError`.

This situation arises in practice after spatial queries or filters that produce no results for a particular region of interest — the element exists in the SpatialData object but contains no valid geometries. The user gets no hint about what went wrong or how to fix it.

---

## Minimal reproducible example

```python
import matplotlib; matplotlib.use("Agg")
import matplotlib.pyplot as plt
import geopandas as gpd
from shapely.geometry import Polygon
import dask; dask.config.set({"dataframe.query-planning": False})
import spatialdata as sd
from spatialdata.models import ShapesModel
import spatialdata_plot

# All shapes are empty geometries
shapes = ShapesModel.parse(gpd.GeoDataFrame({"geometry": [Polygon()]}))
sdata = sd.SpatialData(shapes={"s": shapes})

fig, ax = plt.subplots()
sdata.pl.render_shapes("s").pl.show(ax=ax)
# AssertionError: Cannot compute extent of an empty collection of geometries.
```

---

## Expected behaviour

Either:
- A `UserWarning` is emitted ("Element 's' contains no valid geometries; skipping render") and the plot shows an empty axes
- Or a clear `ValueError` with a diagnostic message explaining the issue

## Actual behaviour

```
AssertionError: Cannot compute extent of an empty collection of geometries.
```

A bare `AssertionError` with no indication of which element is empty or what the user should do.

---

## Fix sketch

Wrap the `get_extent` call at `basic.py:1328` to catch the upstream `AssertionError`:

```python
try:
    extent = get_extent(sdata, coordinate_system=cs, ...)
except (AssertionError, ValueError) as e:
    if "empty" in str(e).lower():
        logger.warning(
            f"All elements to render in coordinate system '{cs}' are empty geometries. "
            "Nothing will be plotted."
        )
        continue  # skip this coordinate system
    raise
```

Alternatively, validate that at least one non-empty geometry exists before calling `get_extent`.

---
**Triage tier**: Tier 3

--- a/src/spatialdata_plot/pl/basic.py
+++ b/src/spatialdata_plot/pl/basic.py
@@ -1513,6 +1513,18 @@ class PlotAccessor:
                 if fig_params.frameon is False:
                     ax.axis("off")
 
+            if has_shapes and wants_shapes:
+                empty_shape_elements = [
+                    name
+                    for name in wanted_elements
+                    if name in sdata.shapes and not sdata.shapes[name]["geometry"].apply(lambda g: not g.is_empty).any()
+                ]
+                if empty_shape_elements:
+                    raise ValueError(
+                        f"Cannot render shape element(s) {empty_shape_elements} in coordinate system {cs!r}: "
+                        "all geometries are empty. Drop the element or restore at least one non-empty geometry."
+                    )
+
             extent = get_extent(
                 sdata,
                 coordinate_system=cs,

